samples/js/photos/src/photo-grid-item.js

```diff
@@ -43,53 +50,96 @@ export default class PhotoGridItem extends React.Component<void, Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
-      size: props.gridSize,
-      sizeIsBest: false,
-      url: props.url,
+      blob: this.getBlob(props),
+      tx: false,
     };
     this._parentTop = 0;
     this._parentLeft = 0;
-    this._shouldTransition = true;
+    this.load(this.state.blob);
   }
 
-  componentWillUpdate() {
+  componentWillReceiveProps(nextProps) {
+    const nextBlob = this.getBlob(nextProps);
+    if (nextBlob == this.state.blob) { 
+      return;
+    }
+    this.setState({
+      blob: nextBlob,
+    });
+    this.load(nextBlob);
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    if (this.state.url != nextState.url) {
+      this.releaseBlobURL();
+    }
+
     const rect = ReactDOM.findDOMNode(this).parentElement.getBoundingClientRect();
     this._parentTop = rect.top;
     this._parentLeft = rect.left;
   }
 
+  getBlob(props: Props): NomsBlob {
+    let [w, h] = props.fullscreen ?
+      [props.viewport.clientWidth, props.viewport.clientHeight] :
+      [props.gridSize.width, props.gridSize.height];
+    const [_, blob] = props.photo.getBestSize(w, h);
+    return blob;
+  }
+
+  async load(blob: NomsBlob) {
+    const b = await blob.targetValue(this.props.db);
```

Comment:
There is no `targetValue` on `NomsBlob`, did this used to be a ref at some point?